src/main/java/de/dataelementhub/model/handler/element/ElementPathHandler.java

```diff
@@ -0,0 +1,134 @@
+package de.dataelementhub.model.handler.element;
+
+import static de.dataelementhub.dal.jooq.Tables.SCOPED_IDENTIFIER;
+import static de.dataelementhub.dal.jooq.Tables.SCOPED_IDENTIFIER_HIERARCHY;
+
+import de.dataelementhub.dal.jooq.enums.Status;
+import de.dataelementhub.dal.jooq.tables.pojos.ScopedIdentifier;
+import de.dataelementhub.model.DaoUtil;
+import de.dataelementhub.model.dto.element.Element;
+import de.dataelementhub.model.dto.element.ElementPath;
+import de.dataelementhub.model.dto.element.section.Identification;
+import de.dataelementhub.model.handler.element.section.DefinitionHandler;
+import de.dataelementhub.model.handler.element.section.IdentificationHandler;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.jooq.CloseableDSLContext;
+
+public class ElementPathHandler {
+
+  /**
+   * Get all available paths for a given element.
+   */
+  public static List<ElementPath> getElementPaths(CloseableDSLContext ctx,
+      int userId, String urn, String languages) {
+    List<ElementPath> elementPaths = new ArrayList<>();
+    List<String> pathUrnsList = completePaths(ctx, userId, Collections.singletonList(urn));
+    List<String> pathDesignationsList = getDesignations(ctx, pathUrnsList, languages);
+    for (int i = 0; i < pathUrnsList.size(); i++) {
+      ElementPath elementPath = new ElementPath();
+      elementPath.setUrns(pathUrnsList.get(i));
+      elementPath.setDesignations(pathDesignationsList.get(i));
+      elementPaths.add(elementPath);
+    }
+    return elementPaths;
+  }
+
+  /**
+   * Complete a given path until the namespace.
+   */
+  public static List<String> completePaths(CloseableDSLContext ctx, int userId,
+      List<String> startPartialPaths) {
+    List<String> partialPaths = new ArrayList<String>(startPartialPaths);
+    for (int i = 0; i < partialPaths.size(); i++) {
+      String partialPath = partialPaths.get(i);
+      String urn;
+      if (partialPath.toLowerCase().contains("namespace")) {
```

Comment:
There are constants for the element types, e.g. ElementType.NAMESPACE.getLiteral() - better check against those. (with toUpperCase ofc)

Reply:
DONE